src/docs/arch-prefetch.dox

```diff
@@ -0,0 +1,106 @@
+/*! @arch_page arch-prefetch Pre-fetch
+
+@section prefetch_overview What is pre-fetching? When is pre-fetching useful?
+Pre-fetching involves bringing content from the disk into the cache before it is
+required by the application in an attempt to reduce time waiting on I/O. Database
+records need to be in the cache before WiredTiger can make changes to them,
+however, disk operations are expensive and there may be some latency when these
+records are not already in the cache when the application needs them.
+
+Applications access the database in different ways, and so it is not necessarily
+ideal to always enable pre-fetching. Pre-fetching is beneficial when an operation
+has locality of reference and it is reasonably likely that the application will
+access data in a sequential way. When future operations are likely to access data
+that is near the data currently being accessed, WiredTiger can predict which pages
+will be needed soon and pre-emptively read them into the cache.
+
+@section prefetch_algorithm Pre-fetch Algorithm
+1. The main application thread finds candidate pages for pre-fetching as it
+traverses the B-tree.
+2. It populates the pre-fetch queue with these candidate pages.
+3. Pre-fetch worker threads will pop off pages from the pre-fetch queue and read
+pages into the cache.
+
+@subsection prefetch_algorithm_find_pages Finding Candidate Pages
+Pre-fetch looks for potential candidate pages in two places: (1) while walking the
+B-tree (\c __tree_walk_internal) and (2) when verifying a tree (\c __verify_tree).
+The \c __wt_session_prefetch_check function contains a few checks to decide if
+pre-fetching should be performed for a given \c ref. Pre-fetching does not operate
+on internal pages, and should not be performed until we have triggered more than
+one leaf page read from sequential read requests. Pre-fetching is also disabled for
+tiered tables, and special operations with the exception of verify.
+
+@subsection prefetch_algorithm_populate_queue Populating the Pre-fetch Queue
+The pre-fetch queue is a first-in-first-out (FIFO) data structure containing the
+page references of potential candidates the system has identified for pre-fetching.
+The \c __wti_btree_prefetch function is responsible for pushing candidate pages
+onto the queue. Given a \c ref, it will queue the next set of pages if they are leaf
+pages, not already in the cache, and not already on the pre-fetch queue. To prevent
+overloading the pre-fetch queue, there is a maximum number of candidate pages allowed
+in the queue (\c WT_MAX_PREFETCH_QUEUE), and a limit on the maximum number of pages
+we can add to the queue during a pre-fetch check (\c WT_PREFETCH_QUEUE_PER_TRIGGER).
+Pages will not be added to the queue if either of these conditions are met.
+
+@subsection prefetch_algorithm_process_pages Pre-fetching Content Into the Cache
```

Comment:
Nitpicky: I think the subsections may need to follow the same capitalization style, either title case (capitalizing the first letter of each significant word) or sentence case (capitalizing only the first letter of the first word). There're some inconsistency, e.g. here we are using title case but "@subsection prefetch_other_cache_busyness Considering current cache busyness" is using sentence case.